lab_6_pipeline/pipeline.py

```diff
@@ -144,16 +254,72 @@ def __init__(self, corpus_manager: CorpusManager):
         """
         Initializes MorphologicalAnalysisPipeline
         """
+        self._corpus = corpus_manager
+        self._mystem = Mystem()
+        mapping_path = Path(__file__).parent / 'data' / 'mystem_tags_mapping.json'
+        self._converter = MystemTagConverter(mapping_path)
 
     def _process(self, text: str) -> List[ConlluSentence]:
         """
         Returns the text representation as the list of ConlluSentence
         """
+        analysis = [token for token in self._mystem.analyze(text) if token['text'].isalnum() or '.' in token['text']]
+        sentence_list = list()
+        for sentence_position, sentence in enumerate(split_by_sentence(text)):
+            sentence_tokens = list()
+            conllu_tokens = list()
+            for token_position, token in enumerate(analysis):
+                if token['text'].strip() in sentence:
+                    sentence_tokens.append(token)
+                else:
+                    del analysis[:token_position]
+                    break
+            token_position = 0
+            for token in sentence_tokens:
+                if 'analysis' in token and token['analysis']:
+                    token_position += 1
+                    conllu_token = ConlluToken(text=token['text'], position=token_position)
+                    morph_params = MorphologicalTokenDTO(lemma=token['analysis'][0]['lex'],
+                                                         pos=self._converter.convert_pos(tags=token['analysis'][0]['gr']),
+                                                         tags=self._converter.convert_morphological_tags(tags=token['analysis'][0]['gr']))
+                    conllu_token.set_morphological_parameters(morph_params)
+                    conllu_tokens.append(conllu_token)
+                elif 'analysis' in token:
+                    token_position += 1
+                    conllu_token = ConlluToken(text=token['text'], position=token_position)
+                    morph_params = MorphologicalTokenDTO(lemma=token['text'],
+                                                         pos='X')
+                    conllu_token.set_morphological_parameters(morph_params)
+                    conllu_tokens.append(conllu_token)
```

Comment:
all elifs can be simplified - look at similar lines

Reply:
Done